Posix Implementation of Raw Time Missing Cast

| | |
|:---|:---|
|**_F´ Version_**| |
|**_Affected Component_**|   |
---
## Problem Description

A description of the problem with sufficient detail to understand the issue.

## Context / Environment

Execute `fprime-util version-check` and share the output.

```
--INSERT HERE--
```

## How to Reproduce

```
.../lib/fprime/Os/Posix/RawTime.cpp:70:39: error: non-constant-expression cannot be narrowed from type 'U32' (aka 'unsigned int') to 'long' in initializer list [-Wc++11-narrowing]
    this->m_handle.m_timespec = {sec, nsec};
                                      ^~~~
.../lib/fprime/Os/Posix/RawTime.cpp:70:39: note: insert an explicit cast to silence this issue
    this->m_handle.m_timespec = {sec, nsec};
                                      ^~~~
                                      static_cast<long>( )
```

We should cast to `time_t` and `long` respectively.

--- a/Os/Posix/RawTime.cpp
+++ b/Os/Posix/RawTime.cpp
@@ -67,7 +67,7 @@ Fw::SerializeStatus PosixRawTime::deserialize(Fw::SerializeBufferBase& buffer) {
     if (status != Fw::SerializeStatus::FW_SERIALIZE_OK) {
         return status;
     }
-    this->m_handle.m_timespec = {sec, nsec};
+    this->m_handle.m_timespec = {static_cast<time_t>(sec), static_cast<long>(nsec)};
     return Fw::SerializeStatus::FW_SERIALIZE_OK;
 }
 

